projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DeploymentServiceV2.java

```diff
@@ -46,82 +42,75 @@ public class DeploymentServiceV2 {
    * configuration; otherwise, a new CronJob will be created. The method ensures that the CronJob in
    * the cluster matches the desired state specified in the configuration.
    *
-   * @param dataJob The data job to update or create.
+   * @param dataJob the data job to be deployed.
+   * @param desiredJobDeployment the desired data job deployment to be deployed.
+   * @param actualJobDeployment the actual data job deployment has been deployed.
+   * @param isJobDeploymentPresentInKubernetes if it is true the data job deployment is present in
+   *     Kubernetes.
    * @param sendNotification if it is true the method will send a notification to the end user.
-   * @param dataJobDeploymentNames list of actual data job deployment names returned by Kubernetes.
    */
   public void updateDeployment(
-      DataJob dataJob, Boolean sendNotification, Set<String> dataJobDeploymentNames) {
-    DataJobDeployment dataJobDeployment = dataJob.getDataJobDeployment();
-
-    if (dataJobDeployment == null) {
+      DataJob dataJob,
+      DesiredDataJobDeployment desiredJobDeployment,
+      ActualDataJobDeployment actualJobDeployment,
+      boolean isJobDeploymentPresentInKubernetes,
+      Boolean sendNotification) {
+    if (desiredJobDeployment == null) {
       log.debug(
           "Skipping the data job [job_name={}] deployment due to the missing deployment data",
           dataJob.getName());
       return;
     }
 
-    JobDeployment jobDeployment =
-        DeploymentModelConverter.toJobDeployment(
-            dataJob.getJobConfig().getTeam(), dataJob.getName(), dataJobDeployment);
-
     try {
-      log.info("Starting deployment of job {}", jobDeployment.getDataJobName());
-      deploymentProgress.started(dataJob.getJobConfig(), jobDeployment);
+      log.info("Starting deployment of job {}", desiredJobDeployment.getDataJobName());
```

Comment:
As stated in a previous comment - we should check if the current cronjob is the same as the desired stated and update the cron job only if there's  a difference.

Reply:
Look at my previous comment.